### PR TITLE
[WIP] Wallet Storage

### PR DESCRIPTION
**Pull request by vimmerru**
Fix before merge to master
- [x] system library dependence for updated libsodium
- [ ] ~restore iOS test for plugged wallet~
- [ ] remove `abort`-like code - will be moved to separate task
- [x] restore wrappers tests

### CODE REVIEW

File: libindy/Cargo.toml [hunk 1]
```diff
@@ -53,10 +53,12 @@ serde_derive = "1.0"
 sha2 = "0.6.0"
 sha3 = "0.6.0"
 rmp-serde = "0.13.6"
-sodiumoxide = {version = "0.0.14", optional = true}
+sodiumoxide = { git="https://github.com/ccfb3ee765a58cae/sodiumoxide.git", branch="xchacha20-poly1305-support", optional = true }
```
Comment (jovfer) [inline libindy/Cargo.toml]: Should not be merged to master.

File: libindy/Cargo.toml [hunk 2]
```diff
@@ -53,10 +55,12 @@ serde_derive = "1.0"
 sha2 = "0.6.0"
 sha3 = "0.6.0"
 rmp-serde = "0.13.6"
-sodiumoxide = {version = "0.0.14", optional = true}
+sodiumoxide = {version = "0.0.16", optional = true}
```
Comment (jovfer) [inline libindy/Cargo.toml]: This update may be problimatic for some platforms

Comment (jovfer) [reply libindy/Cargo.toml]: #653

**Comment by vimmerru** [comment]
@dhh1128 

> Also, I want to eliminate the relationship between wallets and pools. A wallet should be able to hold info from multiple pools, with the DID method prefix guaranteeing uniqueness (that is, we store "did:sov:abcxyz", not just "abcxzy" as the wallet key.

I suggest to implement this as a dedicated PR. Too much for one PR.

**Pull request merged**